src/Domain/Creators/Cylinder.hpp

```diff
@@ -278,6 +279,15 @@ class Cylinder : public DomainCreator<3> {
   std::vector<std::array<size_t, 3>> initial_refinement_levels()
       const noexcept override;
 
+  std::vector<std::string> block_names() const noexcept override {
```

Comment:
why not return `const &` of the names and groups? Returning by value goes against one of the age-old C++ idioms

Reply:
Can I? On Will's request I didn't add protected member variables to the `DomainCreator` base class that get filled in the derived classes and returned by const-ref. Instead, each derived class defines their own member var, and overrides a virtual function. The virtual function returns by value because the base class doesn't know about the member vars, and if a return by const-ref is possible.

I think this is also why the other functions like `initial_refinement_levels` return by value.

Thoughts on this?

Reply:
I guess if you are returning an empty default in the base class you can't return a reference to a temporary, so maybe that was the issue? 

Since we won't call this function frequently I don't think it's a blocker, I just want to understand the reason for this choice :)

Reply:
That's what I meant with my comment above. Is there anything you want me to change here?

Reply:
Nope, that's fine. Makes sense to me now :)